dd-sdk-android/src/test/kotlin/com/datadog/android/utils/DatadogExt.kt

```diff
@@ -52,9 +53,18 @@ inline fun <reified T : Context> mockContext(
  * (debug or release)
  */
 fun resolveTagName(caller: Any, defaultIfNotDebug: String? = null): String {
-    return if (Datadog.isDebug) {
-        caller.javaClass.simpleName
+    val tag = if (Datadog.isDebug) {
+        val javaClass = caller.javaClass
+        val strippedName = javaClass.simpleName.replace(Regex("(\\$\\d+)+$"), "")
+        if (javaClass.isAnonymousClass) {
+            javaClass.enclosingClass!!.simpleName.replace(Regex("(\\$\\d+)+$"), "")
+        } else if (javaClass.isLocalClass) {
+            javaClass.enclosingClass!!.simpleName + "$" + strippedName
+        } else {
+            strippedName
+        }
     } else {
         defaultIfNotDebug ?: "Datadog"
     }
+    return tag.substring(0, min(tag.length, 23))
```

Comment:
I think you want to do this only for SDK version below N

Reply:
Done